src/main/java/uk/gov/justice/laa/portal/landingpage/controller/MultiFirmUserController.java

```diff
@@ -158,6 +162,99 @@ public String addUserProfilePost(@Valid MultiFirmUserForm multiFirmUserForm, Bin
+     */
+    @GetMapping("/user/delete-profile/{userProfileId}")
+    @PreAuthorize("@accessControlService.canDeleteFirmProfile(#userProfileId)")
+    public String deleteFirmProfileConfirm(@PathVariable String userProfileId, Model model) {
+        Optional<UserProfileDto> optionalUserProfile = userService.getUserProfileById(userProfileId);
+        if (optionalUserProfile.isEmpty()) {
+            throw new RuntimeException("User profile not found.");
+        }
+
+        UserProfileDto userProfile = optionalUserProfile.get();
+        EntraUserDto entraUser = userProfile.getEntraUser();
+
+        // Verify this is a multi-firm user
+        if (entraUser == null || !entraUser.isMultiFirmUser()) {
+            throw new RuntimeException("This operation is only available for multi-firm users.");
+        }
+
+        // Count the number of profiles to prevent deletion of last profile
+        List<UserProfile> allProfiles = userService.getUserProfilesByEntraUserId(UUID.fromString(entraUser.getId()));
+        if (allProfiles.size() <= 1) {
+            model.addAttribute("errorMessage", "Cannot delete the last firm profile. User must have at least one profile.");
+            return "redirect:/admin/users/manage/" + userProfileId;
+        }
+
+        model.addAttribute("userProfile", userProfile);
+        model.addAttribute("user", entraUser);
+        model.addAttribute(ModelAttributes.PAGE_TITLE, "Delete firm access - " + entraUser.getFullName());
+        
+        return "multi-firm-user/delete-profile-confirm";
+    }
+
+    /**
+     * Execute deletion of a firm profile from a multi-firm user
+     */
+    @PostMapping("/user/delete-profile/{userProfileId}")
+    @PreAuthorize("@accessControlService.canDeleteFirmProfile(#userProfileId)")
+    public String deleteFirmProfileExecute(@PathVariable String userProfileId,
+                                          @RequestParam(name = "confirm", required = false) String confirm,
+                                          Authentication authentication,
+                                          RedirectAttributes redirectAttributes,
+                                          Model model) {
+        // Get user profile before deletion
+        Optional<UserProfileDto> optionalUserProfile = userService.getUserProfileById(userProfileId);
+        if (optionalUserProfile.isEmpty()) {
+            throw new RuntimeException("User profile not found.");
+        }
+
+        UserProfileDto userProfile = optionalUserProfile.get();
+        EntraUserDto entraUser = userProfile.getEntraUser();
+        final String firmName = userProfile.getFirm() != null ? userProfile.getFirm().getName() : "Unknown";
+
+        // If user selected "No", redirect back to manage user page
+        if ("no".equals(confirm)) {
+            return "redirect:/admin/users/manage/" + userProfileId;
+        }
+
+        // Validate that "Yes" was selected
+        if (!"yes".equals(confirm)) {
```

Comment:
I think is confirm is no we end up inside this if because of exclamation mark 
what do you think?